rust-runtime/aws-smithy-http-server/src/plugin/mod.rs

```diff
@@ -46,41 +54,39 @@ pub trait Plugin<Protocol, Op, S, L> {
     fn map(&self, input: Operation<S, L>) -> Operation<Self::Service, Self::Layer>;
 }
 
-/// An [`Plugin`] that maps an `input` [`Operation`] to itself.
-pub struct IdentityPlugin;
-
-impl<P, Op, S, L> Plugin<P, Op, S, L> for IdentityPlugin {
-    type Service = S;
-    type Layer = L;
-
-    fn map(&self, input: Operation<S, L>) -> Operation<S, L> {
-        input
+/// An extension trait for [`Plugin`].
+pub trait PluginExt<P, Op, S, L>: Plugin<P, Op, S, L> {
+    /// Stacks another [`Plugin`], running them sequentially.
+    fn stack<Other>(self, other: Other) -> PluginStack<Self, Other>
+    where
+        Self: Sized,
+    {
+        PluginStack::new(self, other)
     }
-}
 
-/// A wrapper struct which composes an `Inner` and an `Outer` [`Plugin`].
-pub struct PluginStack<Inner, Outer> {
-    inner: Inner,
-    outer: Outer,
-}
-
-impl<Inner, Outer> PluginStack<Inner, Outer> {
-    /// Creates a new [`PluginStack`].
-    pub fn new(inner: Inner, outer: Outer) -> Self {
-        PluginStack { inner, outer }
+    /// Filters the application of the [`Plugin`] using a predicate over the
+    /// [`OperationShape::NAME`](crate::operation::OperationShape).
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # use aws_smithy_http_server::{plugin::{Plugin, PluginExt}, operation::{Operation, OperationShape}};
+    /// # struct Pl;
+    /// # struct Op;
+    /// # impl OperationShape for Op { const NAME: &'static str = ""; type Input = (); type Output = (); type Error = (); }
+    /// # impl Plugin<(), Op, (), ()> for Pl { type Service = (); type Layer = (); fn map(&self, input: Operation<(), ()>) -> Operation<(), ()> { input }}
+    /// # let plugin = Pl;
+    /// # let operation = Operation { inner: (), layer: () };
+    /// // Prevents `plugin` from being applied to the `CheckHealth` operation.
+    /// let filtered_plugin = plugin.filter_by_operation_name(|name| name != "com.aws.example#CheckHealth");
```

Comment:
I imagine we'd like customers to prefer using the `NAME` associated constant on the relevant `OperationShape` to perform this kind of operations rather than hard-coding a static string.
Although I realise that makes writing the example somewhat more cumbersome.